packages/functions/e2e/functions.e2e.js

```diff
@@ -51,6 +51,24 @@ describe('functions()', function () {
       const response = await functionRunner();
       response.data.should.equal(region);
     });
+
+    it('accepts passing in a custom url string as first arg to an app', async function () {
```

Comment:
can you add the test I reference on slack for useEmulator to make sure that emulator overrides custom URL?

Reply:
Added, I get a `PERMISSION DENIED` locally. Running in CI for testing

Reply:
Hmm - I would be concerned if CI reported a different result than local and in fact they both (ios and android) fail with permission denied in CI as well (which is good because consistency is good)

I think this can only be tested via jest anyway, there is no way to unset emulator usage is there? There is not for auth anyway - sending empty or null in is not documented to clear the value and in practice discards I think

Reply:
Really weird from the android log

```

03-01 11:22:22.915  2720 11416 I MdiSyncModule: flag handling is skipping irrelevant intent.
03-01 11:22:22.954  2720  3098 W Conscrypt: Could not set socket write timeout: java.net.SocketException: Socket closed
03-01 11:22:22.954  2720  3098 W Conscrypt: 	at com.google.android.gms.org.conscrypt.Platform.setSocketWriteTimeout(:com.google.android.gms@201817023@20.18.17 (040800-311416286):2)
03-01 11:22:22.954  2720  3098 W Conscrypt: 	at com.google.android.gms.org.conscrypt.ConscryptFileDescriptorSocket.setSoWriteTimeout(:com.google.android.gms@201817023@20.18.17 (040800-311416286):0)
03-01 11:22:22.968  2720 11854 W ModuleInitIntentOp: Dropping unexpected action com.google.android.gms.phenotype.COMMITTED
03-01 11:22:22.971  2720  3098 W Conscrypt: Could not set socket write timeout: java.net.SocketException: Socket closed
03-01 11:22:22.971  2720  3098 W Conscrypt: 	at com.google.android.gms.org.conscrypt.Platform.setSocketWriteTimeout(:com.google.android.gms@201817023@20.18.17 (040800-311416286):2)
03-01 11:22:22.971  2720  3098 W Conscrypt: 	at com.google.android.gms.org.conscrypt.ConscryptFileDescriptorSocket.setSoWriteTimeout(:com.google.android.gms@201817023@20.18.17 (040800-311416286):0)
03-01 11:22:23.037  2720 11854 I ChromeSync: [Sync,SyncIntentOperation] Handling the intent: Intent { act=com.google.android.gms.phenotype.COMMITTED flg=0x10 pkg=com.google.android.gms cmp=com.google.android.gms/.chimera.GmsIntentOperationService (has extras) }.
03-01 11:22:23.082  2720 11854 I MdiSyncModule: flag handling is skipping irrelevant intent.
03-01 11:22:23.105  2720 11679 E MS_RegisterService: Exception during register request.
03-01 11:22:23.105  2720 11679 E MS_RegisterService: ciau: PERMISSION_DENIED: The caller does not have permission
03-01 11:22:23.105  2720 11679 E MS_RegisterService: 	at ciat.c(:com.google.android.gms@201817023@20.18.17 (040800-311416286):3)
03-01 11:22:23.105  2720 11679 E MS_RegisterService: 	at sdn.a(:com.google.android.gms@201817023@20.18.17 (040800-311416286):48)
03-01 11:22:23.105  2720 11679 E MS_RegisterService: 	at afgm.a(:com.google.android.gms@201817023@20.18.17 (040800-311416286):12)
03-01 11:22:23.105  2720 11679 E MS_RegisterService: 	at com.google.android.libraries.matchstick.net.SilentRegisterIntentOperation.a(:com.google.android.gms@201817023@20.18.17 (040800-311416286):179)
03-01 11:22:23.105  2720 11679 E MS_RegisterService: 	at com.google.android.libraries.matchstick.net.SilentRegisterIntentOperation.f(:com.google.android.gms@201817023@20.18.17 (040800-311416286):38)
03-01 11:22:23.105  2720 11679 E MS_RegisterService: 	at com.google.android.libraries.matchstick.net.SilentRegisterIntentOperation.onHandleIntent(:com.google.android.gms@201817023@20.18.17 (040800-311416286):164)
03-01 11:22:23.105  2720 11679 E MS_RegisterService: 	at com.google.android.chimera.IntentOperation.onHandleIntent(:com.google.android.gms@201817023@20.18.17 (040800-311416286):2)
03-01 11:22:23.105  2720 11679 E MS_RegisterService: 	at qgf.onHandleIntent(:com.google.android.gms@201817023@20.18.17 (040800-311416286):4)
03-01 11:22:23.105  2720 11679 E MS_RegisterService: 	at cui.run(:com.google.android.gms@201817023@20.18.17 (040800-311416286):5)
03-01 11:22:23.105  2720 11679 E MS_RegisterService: 	at cuh.run(:com.google.android.gms@201817023@20.18.17 (040800-311416286):9)
03-01 11:22:23.105  2720 11679 E MS_RegisterService: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
03-01 11:22:23.105  2720 11679 E MS_RegisterService: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
03-01 11:22:23.105  2720 11679 E MS_RegisterService: 	at java.lang.Thread.run(Thread.java:923)
```

That's the stack that lines up with the timestamps and the string PERMISSION_DENIED, but that is not anything like the stack contents I would expect. I'm searching around :thinking:

Reply:
All the online stuff points to google cloud platform enforcing access permissions on some object and kicking out this inscrutable message when it doesn't have it. https://stackoverflow.com/questions/52248879/cloud-composer-permission-denied-the-caller-does-not-have-permission - not sure how to resolve it, but it does appear (online) to be a firestore-related online permission issue

testing emulator-overrides-custom-domain seems best in jest either way - then no worries about this stuff